### ISSUE TITLE
Change parent classes of some events

### ISSUE THREAD

**Issue by franmomu** [doctrine/doctrine2#9875]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | yes and no

#### Summary

Ref: https://github.com/doctrine/orm/issues/4078 and https://github.com/doctrine/orm/pull/1144

`OnClearEventArgs`, `OnFlushEventArgs`, `PostFlushEventArgs` and `PreFlushEventArgs` can be updated keeping BC and `PreUpdateEventArgs` must be done in `3.x`.

- [x] [Update `OnClearEventArgs`, `OnFlushEventArgs`, `PostFlushEventArgs` and `PreFlushEventArgs`](https://github.com/doctrine/orm/pull/9876)
- [x] [Deprecate `LifecycleEventArgs ` in 2.x](https://github.com/doctrine/orm/pull/9906)
- [ ] Update `PreUpdateEventArgs` in 3.x

I'll create the PRs.

**Comment by derrabus** [comment]
Do you want to continue your work on the 3.0.x branch?

**Comment by franmomu** [comment]
Yeah sure, I tried, but bumped into https://github.com/doctrine/orm/pull/9939
